cpp/src/arrow/util/thread_pool.h

```diff
@@ -591,6 +591,21 @@ typename Fut::SyncType RunSynchronously(FnOnce<Fut(Executor*)> get_future,
   }
 }
 
+template <typename T>
```

Comment:
Comment for this?

Reply:
Added and expanded on the comment on the next overloaded definition to explain the difference